src/map/mob.c

```diff
@@ -2447,14 +2448,12 @@ int mob_dead(struct mob_data *md, struct block_list *src, int type) {
 				intif->broadcast(message, (int)strlen(message)+1, BC_DEFAULT);
 			}
 
-			/* heres the thing we got the feature set up however we're still discussing how to best define the ids,
-			 * so while we discuss, for a small period of time, the list is hardcoded (yes officially only those 2 use it,
-			 * thus why we're unsure on how to best place the setting) */
-			/* temp, will not be hardcoded for long thudu. */
-			// TODO: This should be a field in the item db.
-			if (mvp_sd != NULL
-			 && (it->nameid == ITEMID_GOLD_KEY77 || it->nameid == ITEMID_SILVER_KEY77)) /* for when not hardcoded: add a check on mvp bonus drop as well */
-				clif->item_drop_announce(mvp_sd, it->nameid, md->name);
+			// Official Drop Announce [Jedzkie]
+			if ((id = itemdb->search(it->nameid)) != NULL) {
+				if (mvp_sd != NULL && id->flag.drop_announce) {
```

Comment:
At first better check mvp_sd and drop_announce flag, and only after search item.
Because most items not have drop announce flag and item search much slower than check two flags.